packages/plugin-container-app/src/commands/uninstrument.ts

```diff
@@ -0,0 +1,196 @@
+  private tagClient!: TagsOperations
+
+  public async execute(): Promise<0 | 1> {
+    this.enableFips()
+    const [containerAppsToUninstrument, config, errors] = await this.ensureConfig()
+    if (errors.length > 0) {
+      for (const error of errors) {
+        this.context.stdout.write(renderError(error))
+      }
+
+      return 1
+    }
+
+    this.cred = new DefaultAzureCredential()
+    if (!(await ensureAzureAuth((msg) => this.context.stdout.write(msg), this.cred))) {
+      return 1
+    }
+    this.tagClient = new ResourceManagementClient(this.cred).tagsOperations
+
+    this.context.stdout.write(`${this.dryRunPrefix}🐶 Beginning uninstrumentation of Azure Container App(s)\n`)
+    const results = await Promise.all(
+      Object.entries(containerAppsToUninstrument).map(([subscriptionId, resourceGroupToNames]) =>
+        this.processSubscription(subscriptionId, resourceGroupToNames, config)
+      )
+    )
+    const success = results.every((result) => result)
+    this.context.stdout.write(
+      `${this.dryRunPrefix}🐶 Uninstrumentation completed ${
+        success ? 'successfully!' : 'with errors, see above for details.'
+      }\n`
+    )
+
+    return success ? 0 : 1
+  }
+
+  public async processSubscription(
+    subscriptionId: string,
+    resourceGroupToNames: Record<string, string[]>,
+    config: ContainerAppConfigOptions
+  ): Promise<boolean> {
+    const containerAppClient = new ContainerAppsAPIClient(this.cred, subscriptionId)
+    const results = await Promise.all(
+      Object.entries(resourceGroupToNames).flatMap(([resourceGroup, containerAppNames]) =>
+        containerAppNames.map((containerAppName) =>
+          this.processContainerApp(containerAppClient, config, resourceGroup, containerAppName, subscriptionId)
+        )
+      )
+    )
+
+    return results.every((result) => result)
+  }
+
+  /**
+   * Process an Azure Container App for uninstrumentation.
+   * @returns A promise that resolves to a boolean indicating success or failure.
+   */
+  public async processContainerApp(
+    containerAppClient: ContainerAppsAPIClient,
+    config: ContainerAppConfigOptions,
+    resourceGroup: string,
+    containerAppName: string,
+    subscriptionId: string
+  ): Promise<boolean> {
+    try {
+      const containerApp = await containerAppClient.containerApps.get(resourceGroup, containerAppName)
+
+      await this.uninstrumentSidecar(containerAppClient, config, resourceGroup, containerApp)
+      await this.removeTags(subscriptionId, resourceGroup, containerApp)
+    } catch (error) {
+      this.context.stdout.write(renderError(`Failed to uninstrument ${containerAppName}: ${formatError(error)}`))
+
+      return false
+    }
+
+    return true
+  }
+
+  public async removeTags(subscriptionId: string, resourceGroup: string, containerApp: ContainerApp): Promise<void> {
+    const updatedTags = {...containerApp.tags}
+    delete updatedTags.service
+    delete updatedTags.env
+    delete updatedTags.version
```

Comment:
Are these always added by us (the instrument command) or do users sometimes set these themselves?